realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java

```diff
@@ -103,6 +103,12 @@ public void run() {
                                 threadAssertionError[0] = e;
                                 unitTestFailed = true;
                             } finally {
+                                try {
+                                    looperTearDown();
+                                } catch (Throwable t) {
+                                    threadAssertionError[0] = t;
```

Comment:
Check if `threadAssertionError[0]` is `null` first. The first exception is more important.